src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -250,10 +302,43 @@ protected virtual IEnumerable<ServiceBusPlugin> DefineServiceBusPlugins()
         /// <param name="cancellationToken">Indicates that the shutdown process should no longer be graceful.</param>
         public override async Task StopAsync(CancellationToken cancellationToken)
         {
+            if ((Settings.Options.TopicSubscription & TopicSubscription.DeleteOnStop) == TopicSubscription.DeleteOnStop)
+            {
+                await DeleteTopicSubscriptionAsync(cancellationToken);
+            }
+
             await base.StopAsync(cancellationToken);
             _isHostShuttingDown = true;
         }
 
+        private async Task DeleteTopicSubscriptionAsync(CancellationToken cancellationToken)
```

Comment:
Thinking out loud - We could move this to a `AzureServiceBusTopicManager` or so, but not a must.

Reply:
If so, a follow-up PR for this might be better maybe 🤔

Reply:
If it would delete the topic on dispose, I would make it a `IDisposable`; but that's not the case. So, hmm, maybe just too little functionality to get the added value to move this?

Reply:
Would be seperation of concerns but it's ok